### PR TITLE
Build/fuseki deployment

### PR DESCRIPTION
**Pull request by ccamel**
## Purpose

This PR adds three new goals to the `Makefile` for managing a ontology deployments in a local environment. The triple store used is [Apache Jena Fuseki](https://jena.apache.org/documentation/fuseki2/).

The goals added are quite clear:

- `fuseki-start`: starts the Fuseki server in a Docker container if not already running, and uploads the ontology and the examples in it.
- `fuseki-stop`: stops the running Fuseki container if it exists.
- `fuseki-log`: displays the logs of the running Fuseki container.

## Note

- [Fuseki](https://jena.apache.org/documentation/fuseki2/) server is configured with `anonymous` access to simplify administration, so all endpoints are available without needing to provide any credentials. This is very useful for testing and development purposes.
- [docker-compose](https://docs.docker.com/compose/) is not use but instead the `docker run` command directly. This approach allows for greater flexibility and customization, and is in line with what we are doing in other projects @okp4.
- Added the goal `check` which checks if all required commands are available in the system.

### CODE REVIEW

**Review by bdeneux — APPROVED** [review]
That seems good ! 👍 But is missing some `sed` commands 🙃

File: Makefile [hunk 1]
```diff
@@ -192,6 +200,47 @@ $(FLG_TSTS): $(DST_TEST)/%.tested.flag: $(SRC_TST)/%.ttl $(wildcard $(SRC_ONT)/*
...
+	  --rm \
+	  -d \
+	  --name ${DEPLOYMENT_FUSEKI_CONTAINER} \
+	  -p ${DEPLOYMENT_FUSEKI_PORT}:${DEPLOYMENT_FUSEKI_PORT} \
+	  -v `pwd`/shiro.ini:/fuseki/shiro.ini \
+	  -e -JVM_ARGS=${DEPLOYMENT_FUSEKI_JVM_ARGS} \
+	  ${DOCKER_IMAGE_FUSEKI}
+	@sleep 1
+	@echo "${COLOR_CYAN}⏱️ waiting for REST API to be ready...${COLOR_RESET}"
+	@timeout ${DEPLOYMENT_FUSEKI_STARTUP_TIMEOUT} sh -c 'until $$(curl --output /dev/null --silent --head --fail http://localhost:${DEPLOYMENT_FUSEKI_PORT}/$$/ping); do \
```
Comment (amimart) [inline Makefile:221]: I think this command is not installed by default on MacOS systems, we may check its presence

**Review by amimart — COMMENTED** [review]
A just have an optional remark, as you want :)

File: Makefile (hunk 1 above)
Comment (ccamel) [reply Makefile:221]: 🤔 good point. I make the change to include [timeout](https://man7.org/linux/man-pages/man1/timeout.1.html) and [md5sum](https://man7.org/linux/man-pages/man1/md5sum.1.html) as well (which was missing too).

**Comment by CedrickArmel** [comment]
@ccamel If I understand correctly Fuseki will play the role that a solution like Neo4J would have played as @amimart said yesterday in the review. If so, why was this solution preferred to neo4j?

File: Makefile (hunk 1 above)
Comment (amimart) [reply Makefile:221]: Both can be installed on MacOS systems with `brew install coreutils`

Comment (ccamel) [reply Makefile:222]: Done. See: https://github.com/okp4/ontology/pull/116/commits/9c10eaddb27f32f5d0c16e40dba1e0a4137ef05c.

Comment (ccamel) [reply Makefile:222]: Done by https://github.com/okp4/ontology/commit/b06a66409911e321ff6765eb2e5ed88d2edba631.

**Review by amimart — APPROVED** [review]

**Comment by ccamel** [comment]
> @ccamel If I understand correctly Fuseki will play the role that a solution like Neo4J would have played as @amimart said yesterday in the review. If so, why was this solution preferred to neo4j?

Good point @CedrickArmel! We considered various options for storing and querying ontology data. Currently, we have chosen to focus on [Apache Jena Fuseki](https://jena.apache.org/documentation/fuseki2/) as our primary option. The reason being that Fuseki simplifies the deployment, publishing, and querying of ontologies, especially through [SPARQL](https://www.w3.org/TR/rdf-sparql-query/).

However, we're open to exploring other options, and [Neo4j](https://neo4j.com/) is one of them. It's worth noting that Fuseki is primarily an RDF store, whereas Neo4j provides a more graph-centric data model. It's not quite the same approach, and we need to study it more deeply.

Don't hesitate if you have any other questions. 😌

**Pull request merged**